### ISSUE TITLE
Proxy Settings in settings.xml

### ISSUE THREAD

**Issue by smil2k** [GoogleContainerTools/jib#1304]
**Description of the issue**:

We would like to create images behind firewall. The proxy server is configured in the settings.xml, but Jib does not interpret this.

**Expected behavior**:

Settings XML would be used if no https.proxyHost is specified

**Steps to reproduce**:

Add <proxies> section into the settings xml. eg:
```
<proxies>
  <proxy>
      <id>normal</id>
      <active>true</active>
      <protocol>http</protocol>
      <host>proxy</host>
      <port>3128</port>
    </proxy>
    <proxy>
      <id>ssl</id>
      <active>true</active>
      <protocol>https</protocol>
      <host>proxy</host>
      <port>3128</port>
    </proxy>
  </proxies>
```
**Environment**: 
mvn 3.6.0
linux ubuntu

**Comment by chanseokoh** [comment]
@smil2k thanks for the feedback.

Making some notes for ourselves:
https://github.com/GoogleContainerTools/jib/issues/525#issuecomment-403847186
https://github.com/GoogleContainerTools/jib/issues/525#issuecomment-403995675

**Comment by smil2k** [comment]
I like to use the below project to mock external servers in unit tests:
https://github.com/rest-driver/rest-driver/tree/master/rest-client-driver

**Comment by chanseokoh** [comment]
Fixed by #1337 and #1366. Jib will start using the proxy settings in Maven `settings.xml`. System properties will still have priority over Maven proxy settings.
